dagshub/upload/wrapper.py

```diff
@@ -194,10 +195,10 @@ def upload(
 
         """
         if os.path.isdir(local_path):
-            dir_to_upload = self.directory(remote_path)
+            dir_to_upload = self.directory(remote_path if remote_path is not None else DEFAULT_REMOTE_PATH)
             dir_to_upload.add_dir(local_path, commit_message=commit_message, **kwargs)
         else:
-            file_to_upload = DataSet.get_file(local_path, remote_path)
+            file_to_upload = DataSet.get_file(local_path, remote_path if remote_path is not None else DEFAULT_REMOTE_PATH)
             self.upload_files([file_to_upload], commit_message=commit_message, **kwargs)
```

Comment:
The `get_file` function already handles the `remote_path is None` case. 
It gets the filename from the `local_path` and uploads with that. So this is not required + it's going to always write the file at `/data/src/`